x-pack/plugins/profiling/public/components/flame_graphs_view/index.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiPageHeaderContentProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { ElasticFlameGraph } from '../../../common/flamegraph';
+import { useProfilingParams } from '../../hooks/use_profiling_params';
+import { useProfilingRouter } from '../../hooks/use_profiling_router';
+import { useProfilingRoutePath } from '../../hooks/use_profiling_route_path';
+import { useTimeRange } from '../../hooks/use_time_range';
+import { FlameGraphContext } from '../contexts/flamegraph';
+import { useProfilingDependencies } from '../contexts/profiling_dependencies/use_profiling_dependencies';
+import { FlameGraph } from '../flamegraph';
+import { FlameGraphNavigation } from '../flamegraph_nav';
+import { ProfilingAppPageTemplate } from '../profiling_app_page_template';
+
+export function FlameGraphsView({ children }: { children: React.ReactElement }) {
+  const {
+    query,
+    query: { rangeFrom, rangeTo, index, projectID, n },
+  } = useProfilingParams('/flamegraphs/*');
```

Comment:
`useProfilingParams` will return typed parameters for the given route. If the component is being rendered but the given is not active, the hook will throw an error.